library/std/src/thread/local.rs

```diff
@@ -217,7 +217,7 @@ macro_rules! __thread_local_inner {
                         //   so now.
                         0 => {
                             $crate::thread::__FastLocalKeyInner::<$t>::register_dtor(
-                                &VAL as *const _ as *mut u8,
+                                $crate::ptr::addr_of_mut!(VAL) as *mut u8,
```

Comment:
Note that `&mut VAL as *mut _ as *mut u8` would *not* be right as the `&mut` here would make a unique pointer which (together with the raw ptrs derived from it) becomes invalid when `&VAL` is executed later.